geode-core/src/test/java/org/apache/geode/internal/JarDeployerIntegrationTest.java

```diff
@@ -144,40 +131,99 @@ public void run() {
       Thread.sleep(500);
       alternateDir.mkdir();
       thread.join();
-    } catch (Exception e) {
-      fail(e);
+    } catch (InterruptedException iex) {
+      fail("Interrupted while waiting.");
```

Comment:
In particular, this is the worst thing we as developers can do in a JUnit Test. This eats the exception and then throws an entirely new Exception stack without the original stack trace. All JUnit @Test methods should simple have "throws Exception" and then avoid this stuff.